python/cudf/udf_cpp/shim.cu

```diff
@@ -388,25 +388,58 @@ __device__ bool are_all_nans(cooperative_groups::thread_block const& block,
   return count == 0;
 }
 
-template <typename T>
-__device__ void device_sum(cooperative_groups::thread_block const& block,
-                           T const* data,
-                           int64_t size,
-                           T* sum)
+template <typename T, std::enable_if_t<std::is_integral_v<T>, int> = 0>
+__device__ int64_t device_sum(cooperative_groups::thread_block const& block,
```

Comment:
Rather than writing an entirely new function, could we have a single kernel and write something like this in a second template parameter:

`typename AccumT = std::conditional<std::is_integral_v<T>, int64_t, T>`

Reply:
yes! was hoping someone would comment with something like this :)

Reply:
![giphy](https://github.com/rapidsai/cudf/assets/3943761/197dde98-6c11-45d0-b1f4-d5143614c430)

Reply:
Should be updated! Looking much cleaner now, thanks.